tcms/static/js/testrun_actions.js

```diff
@@ -259,91 +269,61 @@ function updateRunStatus(object_pk, value, callback) {
   });
 }
 
-var updateCaseRunStatus = function(e) {
-  e.stopPropagation();
-  e.preventDefault();
-  var container = jQ(this).parents().eq(3);
-  var parent = container.parent();
-  var title = parent.prev();
-  var link = title.find('.expandable')[0];
-  var parameters = Nitrate.Utils.formSerialize(this);
-  var object_pk = parameters['object_pk'];
-  var value = parameters['value'];
-
-  // Callback when
-  var callback = function(t) {
-    // Update the contents
-    const value = parameters['value']
-    if (value != '') {
-      // Update the case run status icon
-      var crs = Nitrate.TestRuns.CaseRunStatus;
-      title.find('.execution_status_icon').each(function(index) {
-        for (i in crs) {
-          jQ(this).removeClass(crs[i].icon);
-        }
-        const execution = Nitrate.TestRuns.CaseRunStatus.find(tcs => tcs.pk === value);
-        jQ(this).addClass(execution.icon);
-        jQ(this).css('color', execution.color)
-      });
+function updateExecutionStatus(executionId, caseId, executionStatusPk) {
 
-      // Update related people
-      var usr = Nitrate.User;
-      title.find('.link_tested_by').each(function(i) {
-        jQ(this).html(usr.username);
-      });
-    }
+  const container = jQ('.update_form').parents().eq(3)
+  const parent = container.parent()
+  const title = parent.prev()
+  const link = title.find('.expandable')[0];
+
+  const comment = jQ('td > textarea[name=comment]').val()
+
+  setContainerToLoading(container, caseId)
+
+  if (comment) {
+    jsonRPC('TestExecution.add_comment', [executionId, comment], () => updateCommentsCount(caseId, true));
+  }
+
+  // Update the object when changing the status
+  jsonRPC('TestExecution.update', [executionId, {'status': executionStatusPk}], () => {
+    // Update the case run status icon
+    // TODO: remove this global variable in favour of API call
+    var executionStatuses = Nitrate.TestRuns.CaseRunStatus;
+    title.find('.execution_status_icon').each((_, icon) => {
+      executionStatuses.forEach(status => jQ(icon).removeClass(status.icon))
+
+      const execution = executionStatuses.find(tcs => tcs.pk === executionStatusPk)
+      jQ(icon).addClass(execution.icon)
+      jQ(icon).css('color', execution.color)
+    });
```

Comment:
this entire block doesn't appear to do anything useful ATM. After for example setting a status of PASS I can see the button is still enabled and if pressed again I can see in browser console that another RPC method is sent.

Instead doing all this looping around why not mark the status buttons with executionID (you already have status id) and directly select the button with jQuery making it disabled ?

Reply:
you are looking at a wrong place. What this code is doing is updating the icon on the row
this icon:
<img width="245" alt="Screen Shot 2019-12-11 at 20 07 01" src="https://user-images.githubusercontent.com/18421997/70647523-cd44fa80-1c51-11ea-8212-9eead782ff6d.png">

I agree that currently, no icons are disable, and this is a regression I introduced, but will be another piece of code entirely

Reply:
- then update the comments to make it clear which part of the UI this block of code is updating
- `title.find('.execution_status_icon')` should actually return only 1 element so maybe don't use the forEach loop, not really necessary
- `executionStatuses.forEach` - add comment to explain why we need to do it in this way b/c it's confusing
- `tcs` & `execution` - these are test execution statuses. Name them properly.

Enable/disable of the status buttons under the comment form is a separate issue then

Reply:
> Enable/disable of the status buttons under the comment form is a separate issue then

Should I fix it in another PR or in this one? This one is big enough already, so I prefer to do it in another

Reply:
it is realted to the current changes so must go here.

Reply:
This wasn't resolved - it is now.